addons/website_hr_recruitment/static/src/interactions/website_hr_applicant_form.js

```diff
@@ -25,71 +66,25 @@ publicWidget.registry.hrRecruitment = publicWidget.Widget.extend({
             linkedinProfileEl?.removeAttribute("required");
             resumeEl?.removeAttribute("required");
         }
-    },
+    }
 
-    hideWarningMessage(targetEl, messageContainerId) {
-        targetEl.classList.remove("border-warning");
-        document.querySelector(messageContainerId)?.classList.add("d-none");
-    },
-
-    showWarningMessage(targetEl, messageContainerId, message) {
-        targetEl.classList.add("border-warning");
-        document.querySelector(messageContainerId).textContent = message;
-        document.querySelector(messageContainerId)?.classList.remove("d-none");
-    },
-
-    async _onFocusOutName(ev) {
-        const field = "name"
-        const messageContainerId = "#warning-message";
-        await this.checkRedundant(ev.currentTarget, field, messageContainerId);
-    },
-
-    async _onFocusOutMail (ev) {
-        const field = "email"
-        const messageContainerId = "#warning-message";
-        await this.checkRedundant(ev.currentTarget, field, messageContainerId);
-    },
-
-    async _onFocusOutPhone (ev) {
-        const field = "phone"
-        const messageContainerId = "#warning-message";
-        await this.checkRedundant(ev.currentTarget, field, messageContainerId);
-    },
-
-    async _onFocusOutLinkedin (ev) {
+    onFocusOutLinkedin(ev) {
         const targetEl = ev.currentTarget;
         const linkedin = targetEl.value;
         const field = "linkedin";
         const messageContainerId = "#linkedin-message";
         const linkedin_regex = /^(https?:\/\/)?([\w\.]*)linkedin\.com\/in\/(.*?)(\/.*)?$/;
-        let hasWarningMessage = false;
         if (!linkedin_regex.test(linkedin) && linkedin !== "") {
             const message = _t("The profile that you gave us doesn't seems like a linkedin profile")
-            this.showWarningMessage(targetEl, "#linkedin-message", message);
-            hasWarningMessage = true;
+            this.showWarningMessage(targetEl, messageContainerId, message);
+            this.checkRedundant(targetEl, field, messageContainerId, true);
```

Comment:
`checkRedundant` is async => waitFor

Reply:
There is no need for async / await since it is the last step of the handler.
Moreover `checkRedundant`already does the `waitFor`.